src/components/CourseChecklist/index.jsx

```diff
@@ -1,102 +1,179 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import { Icon } from '@edx/paragon';
 import classNames from 'classnames';
+import { elementType } from 'airbnb-prop-types';
 import FontAwesomeStyles from 'font-awesome/css/font-awesome.min.css';
+import { Hyperlink, Icon } from '@edx/paragon';
+import PropTypes from 'prop-types';
+import React from 'react';
+
 import getFilteredChecklist from '../../utils/CourseChecklist/getFilteredChecklist';
 import getValidatedValue from '../../utils/CourseChecklist/getValidatedValue';
+import messages from './displayMessages';
 import styles from './CourseChecklist.scss';
+import WrappedMessage from '../../utils/i18n/formattedMessageWrapper';
 
 class CourseChecklist extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      headingID: '',
       checks: [],
-      totalChecks: 0,
+      totalCompletedChecks: 0,
       values: {},
     };
   }
 
   componentWillMount() {
     this.updateChecklistState(this.props);
-
-    if (this.props.dataHeading) {
-      this.setState({
-        headingID: `${this.props.dataHeading.split(/\s/).join('-')}-heading`,
-      });
-    }
   }
 
   componentWillReceiveProps(nextProps) {
     this.updateChecklistState(nextProps);
   }
 
+  getCompletionCountID = () => (`${this.props.idPrefix.split(/\s/).join('-')}-completion-count`);
+
   getHeading = () => (
     <h3 className={classNames('font-weight-normal', styles['font-extra-large'])}>
-      <span id={this.state.headingID}>{this.props.dataHeading}</span>
+      <span aria-describedby={this.getCompletionCountID()}>{this.props.dataHeading}</span>
```

Comment:
The `aria-describedby` attribute should be on the header, not the element describing the header, so I have swapped that.

Reply:
Not vital, but could the aria-describedby go onto the `h3` and then the `span` could be removed?